anna/my-app/src/pages/home/HomePage.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import Layout from '../../components/layout';
+import './homePage.css';
+import MovieList from '../../components/movieList';
+import SearchForm from '../../components/searchForm';
+import { ISearchFormState } from '../../components/searchForm/SearchForm';
+import { SortType } from '../../components/sortBox/SortBox';
+import SortBox from '../../components/sortBox';
+import Container from '../../components/container';
+import InfiniteScroll from '../../components/infiniteScroll';
+
+import * as QueryString from "query-string";
+import { RouteComponentProps } from "react-router-dom";
+import { ISearchParams } from '../../Api';
+import { MoviesConnectedProps } from '.';
+
+class HomePage extends React.Component<RouteComponentProps & MoviesConnectedProps> {
+
+  componentDidMount() {
+    console.log('mount')
+    const searchParam = QueryString.parse(this.props.location.search);
+
+    if (searchParam.sortBy) {
+      this.props.changeSortBy(searchParam.sortBy as SortType);
+    }
+    this.changeMoviesByPath();
+  }
+
+
+  componentDidUpdate(prevProps: RouteComponentProps & MoviesConnectedProps) {
+    const { location } = this.props;
+
+    if (prevProps.location.search !== location.search) {
+      this.changeMoviesByPath();
+    }
+  }
+
+  changeMoviesByPath = async (): Promise<void> => {
+    const { location } = this.props;
+    const searchParams: ISearchParams = QueryString.parse(`?sortBy=${this.props.sortBy}`)
+
+    if (location.pathname === "/") {
+      this.props.fetchMoviesRequested(searchParams)
+
+    } else if (location.pathname === "/search") {
+      this.props.fetchMoviesRequested(QueryString.parse(location.search))
+    }
+  }
+
+  handleFormSubmit = (searchParams: ISearchFormState) => {
+    const { history } = this.props;
+    const searchQuery = QueryString.stringify(searchParams);
+    history.push({
+      pathname: '/search',
+      search: `${searchQuery}`,
+    });
+  }
+
+  handleSearchChange = (search: ISearchFormState) => {
+    const { history } = this.props;
+    const searchQuery = QueryString.stringify(search);
+
+    history.push({
+      search: searchQuery,
+    });
+  }
```

Comment:
These are two identical methods!  And are you sure that you don't replace SORT_BY if it was already in the URL ?